### PR TITLE
LPS-168726 Remove TemplateManagerUtil manual lifecycle methods and use Component lifecycle methods instead

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

**Comment by zhangyan0701** [comment]
Portal startup error:
```
[beanshell] 2022-11-17 02:53:28.673 ERROR [main][velocity:151] Velocimacro : Velocimacro : Error using VM library : com.liferay.portal.template.velocity_6.0.31_CLASS_LOADER_CONTEXT_VM_global_library.vm
[beanshell] 2022-11-17 02:53:28.674 ERROR [main][velocity:151] org.apache.velocity.exception.ResourceNotFoundException: Unable to find Velocity template with ID com.liferay.portal.template.velocity_6.0.31_CLASS_LOADER_CONTEXT_VM_global_library.vm
```

Full console log:
https://testray.liferay.com/reports/production/logs/2022-11/test-1-29/test-portal-acceptance-pullrequest(master)/5459/functional-smoke-tomcat90-mysql57-jdk8_stable/0/0/jenkins-console.txt.gz?authuser=0

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
pushed with additional commit: @julschong
[LPS-168726 Delay VelocityManager start timing](https://github.com/liferay-core-infra/liferay-portal/pull/2466/commits/5d12b5317d50baf74f89f359889c7aea936bd182)

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by tinatian** [comment]
@julschong 
1. Please move the logic of init and destroy to activate/deactivate first, then make FreemarkerManagerUtil to use service tracker, or else if you are on this commit https://github.com/liferay-core-infra/liferay-portal/pull/2466/commits/8ef7508811aee1f8b2025b19d4f28fdaf757bd6a, your portal will not work well as no one started template manager.
2. The service tracker map you opened is not closed

File: modules/apps/portal-template/portal-template-velocity/src/main/java/com/liferay/portal/template/velocity/internal/VelocityManager.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
  */
 @Component(
 	configurationPid = "com.liferay.portal.template.velocity.configuration.VelocityEngineConfiguration",
-	configurationPolicy = ConfigurationPolicy.OPTIONAL, immediate = true,
+	configurationPolicy = ConfigurationPolicy.OPTIONAL,
```
Comment (tinatian) [inline modules/apps/portal-template/portal-template-velocity/src/main/java/com/liferay/portal/template/velocity/internal/VelocityManager.java:52]: @julschong Do not fix it this way, let's figure out the right dependencies to fix it in the root.

**Comment by tinatian** [comment]
@julschong Please see my comment and resend. 

Thanks
Tina

**Comment by julschong** [comment]
@tinatian Hi Tina,
Some of my questions below:
> 1. Please move the logic of init and destroy to activate/deactivate first, then make **FreemarkerManagerUtil** to use service tracker, or else if you are on this commit [8ef7508](https://github.com/liferay-core-infra/liferay-portal/commit/8ef7508811aee1f8b2025b19d4f28fdaf757bd6a), your portal will not work well as no one started template manager.

TemplateManagerUtil already was using a service tracker, but it was not using service tracker map. Instead it was using a ServiceTrackerCustomizer to manually update the _templateManagers hashmap.

> 2. The service tracker map you opened is not closed

I'm not sure where and when I should close the map as TemplateManagerUtil is in portal-kernel. Most serviceTrackMap.close() I find in the codebase are in deactivate method in a component

Thank you!

**Pull request closed without merging**